autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -602,6 +617,37 @@ def __init__(
 
         Returns:
             None
+
+        Example:
+            RetrieveUserProxyAgent(
+                name="ragproxyagent",
+                human_input_mode="NEVER",
+                max_consecutive_auto_reply=3,
+                retrieve_config={
+                    "task": "code",
+                    "docs_path": [
+                        "https://raw.githubusercontent.com/microsoft/FLAML/main/website/docs/Examples/Integrate%20-%20Spark.md",
+                        "https://raw.githubusercontent.com/microsoft/FLAML/main/website/docs/Research.md",
+                        "https://raw.githubusercontent.com/Knuckles-Team/geniusbot/main/README.md",
+                        "https://raw.githubusercontent.com/Knuckles-Team/repository-manager/main/README.md",
+                        "https://raw.githubusercontent.com/Knuckles-Team/gitlab-api/main/README.md",
+                        "https://raw.githubusercontent.com/Knuckles-Team/media-downloader/main/README.md",
+                        os.path.join(os.path.abspath(""), "..", "website", "docs"),
+                    ],
+                    "custom_text_types": ["non-existent-type"],
+                    "chunk_token_size": 2000,
+                    "model": config_list[0]["model"],
+                    "vector_db": "pgvector",  # PGVector database
+                    "collection_name": "test_collection",
+                    "db_config": {
+                        "connection_string": "postgresql://postgres:postgres@localhost:5432/postgres",
+                    },
+                    "embedding_function": "all-distilroberta-v1",
```

Comment:
I have updated these references to use the embedding function only. No string inputs.

Reply:
> embedding_function should be a callable. In your code, it's a class which has `encode` method.
> 
> To be consistent with the protocol, it should support direct call. Say `emb_func`, then `emb_func(text)` should return the embedding vector or list.

I could agree with this. I encountered issue when using azure openai embedding function because it needs to call encode method which not exists. i posted the details in #2813

Reply:
@Knucklessg1 A better embedding_function would support `embeddings = self.embedding_function(sentences)` instead of `embeddings = self.embedding_function.encode(sentences)`. Which means,  in your case, `self.embedding_function = SentenceTransformer("all-MiniLM-L6-v2").encode` instead of `self.embedding_function = SentenceTransformer("all-MiniLM-L6-v2")`